tests/testthat/_snaps/backend-teradata.md

```diff
@@ -53,9 +53,21 @@
       mf %>% group_by(y) %>% mutate(rown = row_number())
     Output
       <SQL>
-      SELECT `df`.*, ROW_NUMBER() OVER (PARTITION BY `y` ORDER BY `y`) AS `rown`
+      SELECT
+        `df`.*,
+        ROW_NUMBER() OVER (PARTITION BY `y` ORDER BY (SELECT NULL)) AS `rown`
```

Comment:
Interestingly, Teradata (via `win_rank_tdata()`) was previously defaulting to `ORDER BY` the result of `win_current_group()` (added in https://github.com/tidyverse/dbplyr/commit/4109051ff697446bc5a280d65aa708e568047f57#diff-c4f086980e5692d16e3337acd7bfefe3a9315135688133ee4f24ea2944ccba3bR174 by @overmar ).

By generalizing the behavior to  fall back to `SELECT NULL` for all back ends (incl. Snowflake / MSSQL) this now no longer defaults to ordering by the group. 

I'm likely not familiar enough with (1) PARTITION vs ORDER  and (2) Teradata to determine whether this would have adverse impacts, so would welcome input from more knowledgeable folks!